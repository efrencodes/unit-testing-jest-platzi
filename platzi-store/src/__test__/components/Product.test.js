import React from 'react';
import ProviderMock from '../../__mocks__/ProviderMock';
import ProductMock from '../../__mocks__/ProductMock';
import { mount, shallow } from 'enzyme';
import Product from '../../components/Product';
import { create } from 'react-test-renderer';

describe('Componente <Product />', () => {
  test('Render del componente Product', () => {
    const product = shallow(
      <ProviderMock>
        <Product />
      </ProviderMock>
    );
    expect(product.length).toEqual(1);
  });
  test('Comprobar el botÃ³n de comprar', () => {
    const handleAddToCart = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <Product product={ProductMock} handleAddToCart={handleAddToCart} />
      </ProviderMock>
    );
    wrapper.find('button').simulate('click');
    expect(handleAddToCart).toHaveBeenCalledTimes(1);
  });
});

describe('Product Snapshot', () => {
  const handleAddToCart = jest.fn();
  test('Comprobar la UI del componente Product', () => {
    const product = create(
      <ProviderMock>
        <Product product={ProductMock} handleAddToCart={handleAddToCart} />
      </ProviderMock>
    );
    expect(product.toJSON()).toMatchSnapshot();
  });
});
