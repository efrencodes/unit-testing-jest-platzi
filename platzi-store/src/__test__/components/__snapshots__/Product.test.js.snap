// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Snapshot Comprobar la UI del componente Product 1`] = `
<div
  className="Products-item"
>
  <img
    alt="Camiseta"
    src="https://arepa.s3.amazonaws.com/camiseta.png"
  />
  <div
    className="Products-item-info"
  >
    <h2>
      Camiseta
      <span>
        $
        25
      </span>
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit
    </p>
  </div>
  <button
    type="button"
  >
    Comprar
  </button>
</div>
`;
